src/algebra/group_with_zero/defs.lean

```diff
@@ -45,6 +45,24 @@ mul_zero_class.zero_mul a
 @[ematch, simp] lemma mul_zero (a : M₀) : a * 0 = 0 :=
 mul_zero_class.mul_zero a
 
+variables {ι : Type*} [decidable_eq ι]
+
+lemma pi.single_mul_fun_apply (i j : ι) (a : M₀) (f : ι → M₀) :
+  pi.single i a j * f j = pi.single i (a * f i) j :=
+pi.apply_single (λ i, (* f i)) (λ i, zero_mul _) _ _ _
+
+lemma pi.fun_mul_single_apply (i j : ι) (f : ι → M₀) (a : M₀) :
+  f j * pi.single i a j = pi.single i (f i * a) j :=
+pi.apply_single (λ i, ((*) (f i))) (λ i, mul_zero _) _ _ _
+
+lemma pi.single_mul_fun (i : ι) (a : M₀) (f : ι → M₀) :
+  pi.single i a * f = pi.single i (a * f i) :=
+funext $ λ j, pi.single_mul_fun_apply i j a f
+
+lemma pi.fun_mul_single (i : ι) (f : ι → M₀) (a : M₀) :
+  f * pi.single i a = pi.single i (f i * a) :=
+funext $ λ j, pi.fun_mul_single_apply i j f a
```

Comment:
`fun` seems odd to me. Would `single_mul_left` and `single_mul_right` be better names in your opinion? I don't know which way around they'd go, but I assume there's some prior art we can copy, maybe about multiplication by equiv.perm.swap.

Reply:
I did like the names "mul_single" and "single_mul" because it was clear which one is left/right, same here but indeed `fun` seems odd. Actually, I can just remove these lemmas from this PR. It turns out that I won't need them for my current goals, so I'm ready to leave this decision to the first person who'll actually need these lemmas.

Reply:
I think having the lemmas with the wrong name is better than not having them

Reply:
Feel free to adopt #6418